test/langtools/jdk/jshell/ToolingTest.java

```diff
@@ -79,4 +79,43 @@ public void testDisassembleNewRecordClass() {
                 )
         );
     }
+
+    @Test
+    public void testDisassembleBuiltinInnerClass() {
+        test(
+            a -> assertCommand(a, "/open TOOLING",
+                        ""),
+            a -> assertCommandUserOutputContains(a, "javap(Base64.Decoder.class)",
+                        "Classfile jrt:/java.base/java/util/Base64$Decoder.class",
+                        "class java.util.Base64$Decoder",
+                        "SourceFile: \"Base64.java\"")
+        );
+    }
+
+    @Test
+    public void testDisassembleBuiltinAnonymousClass() {
+        test(
+            a -> assertCommand(a, "/open TOOLING",
+                        ""),
+            a -> assertCommandUserOutputContains(a, "javap(Class.forName(\"sun.util.PreHashedMap$1\"))", // arbitrary built-in anonymous class
```

Comment:
I recommend the case to test against a generic class without a canonical name, such as `java.lang.invoke.DirectMethodHandle$Holder` generated by jlink; arbitrary javac artifacts is subject to renames because they have no linkage exposure, and these generic packages with antique code may be removed in a future maintenance and cause unexpected failures.

Reply:
`java.lang.invoke.DirectMethodHandle$Holder` seems to have a canonical name, it's not anonymous:

```
jshell> Class.forName("java.lang.invoke.DirectMethodHandle$Holder").getCanonicalName()
$13 ==> "java.lang.invoke.DirectMethodHandle$Holder"
jshell> Class.forName("java.lang.invoke.DirectMethodHandle$Holder").isAnonymousClass()
$14 ==> false
```

Wasn't too sure on this test either way, considering I also added a test for user created anonymous classes, so I think I'll just remove it instead. Any anonymous class can be refactored anytime and would have the same problem.